Split client views up, aded change_password, and approve posts

thequbit opened the pull request:
This is, admittedly, a very large PR.  Things just got sloppy with the format of client views and admin views.  This is the same treatment that the admin views got [here](https://github.com/thequbit/yellr-server/commit/25ff707ad5ce41df7d30485a41c20c16293c33b2).

This new method of breaking things up matches the new documentation structure [here](https://github.com/hhroc/yellr-server/wiki/Client-APIs-Table-of-Contents).

Tests are still not where I want them to be, but everything does appear to be working as expected.  Tested from the emulator, app, and I walked through the functions of the moderator site as well.

Nolski commented on tests.py:
```diff
@@ -601,7 +601,7 @@ def run_tests():
 
     success, payload = _execute_test(
-        'admin/get_user_posts.json',
+        'admin/get_client_posts.json',
```
am i going to have to worry about this route changing in the front end code?

thequbit replied:
You will when you implement it, however it did not look like you had done anything with the client posts yet.  This would be an end point that would be called to get posts just from a specific user.

Nolski commented on yellr-serv/yellrserv/ep_client_message.py:
```diff
@@ -0,0 +1,115 @@
+            result['error_text'] = "Message already has posted response."
+
+    except:
+        pass
```
Does this need to throw an exception? Anytime `except: pass` is shown there should be a comment saying why it isn't throwing an exception.

thequbit replied:
I appreciate the thoroughness, I'll fix this and add it to the PR.

thequbit commented:
Fixed issue with create_response_message.json.

https://github.com/hhroc/yellr-server/pull/180/files#diff-dacb695c58f252e4618090e8d9c8c258R68

@Nolski let me know if you catch anything else.

Nolski commented on yellr-serv/yellrserv/ep_client_post.py:
```diff
@@ -0,0 +1,244 @@
+                start = int(float(request.GET['start']))
+            if 'count' in request.GET:
+                count = int(float(request.GET['count']))
+        except:
```
I think it might be a good idea to grep for
```python
except:
    pass
```
because I'm still seeing these pop up without any comments or anything like that.

thequbit replied:
in this case, the try/except is catching an invalid number within the start or count GET params.  We set defaults for each, so there is no reason to do anything with the exception.  I see no reason to not do it this way, however if you do not want to see it, the start and count defaults could be placed within the except.

I think this is a completely acceptable way to do this though.

Nolski replied:
I'm not quite sure that I understand. If they give us an invalid number (which is an invalid user input), then that should definitely be caught and in this case throw a 403 to the user saying the input it incorrect.

thequbit replied:
The thought process was that if they did not supply a start and count, then it would default to 0 and 50.  The thought was that the fields would be optional.

I do agree that, if they do supply them and then they are invalid, then it should be 403.  I'll adjust the logic.

Nolski commented on yellr-serv/yellrserv/ep_client_profile.py:
```diff
@@ -0,0 +1,101 @@
+    return utils.make_response(result, status_code)
+
+@view_config(route_name='verify_client.json')
+def verify_user(request):
```
should this endpoint exist if it isn't implemented yet?

thequbit replied:
I was keeping it in there so I could then implement it after this merge, but I'll remove it and then re-add.

Nolski commented on yellr-serv/yellrserv/ep_client_story.py:
```diff
@@ -0,0 +1,59 @@
+    status_code = 200
+
+    try:
+    #if True:
```
are any of these `#if True:` statements worth having? I think if we have exceptions being thrown we should know which blocks are breaking.

thequbit replied:
Removed with latest push to thequbit/dev

thequbit commented:
Cleaned up the code some more and removed extraneous comments.

Note there are still places where additional code clean up can happen, however I think the end points are pretty good right now.  I need to tough the __init__.py file that has all of the routes in it at some point as well, but that can wait.

Nolski commented:
This all looks pretty good to me. I made one small comment on a line above but other than that (if you want to fix that and then merge I'm cool with that as well)

thequbit commented:
final fixes for input sanitation.

@Nolski, merge if you're happy.

Nolski commented:
You da man @thequbit
